File: iHonour/iHonour/Classes/News/AdmobCell.m
//
//  AdmobCell.m
//  iHonour
//
//  Created by Tengfei on 2017/3/27.
//  Copyright © 2017年 ftf. All rights reserved.
//

#import "AdmobCell.h"


@implementation AdmobCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

+(instancetype)cellWithTableview:(UITableView *)tableView{
    static NSString *ID = @"AdmobCell";
    AdmobCell *cell = [tableView dequeueReusableCellWithIdentifier:ID];
    if(cell == nil){
        cell = [[AdmobCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:ID];
    }
    return cell;
}

-(instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        [self initliza];
    }
    return self;
}

-(void)initliza
{
    self.backgroundColor = KColor(239, 239, 244);
    
    GADBannerView *bannerView = [[GADBannerView alloc] initWithAdSize:kGADAdSizeBanner];
    self.bannerView = bannerView;
    self.clipsToBounds = YES;
    //    bannerView.frame = self.bounds;
    [self addSubview:self.bannerView];
    [bannerView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.bottom.left.right.equalTo(self);
    }];
    
    KLog(@"Google Mobile Ads SDK version: %@", [GADRequest sdkVersion]);
    self.bannerView.adUnitID = self.adUnitID;
    self.bannerView.delegate = self;
}

- (void)adViewDidReceiveAd:(GADBannerView *)bannerView{
    self.hidden = NO;
    
    KLog(@"--adViewDidReceiveAd-:successs");
}
- (void)adView:(GADBannerView *)bannerView didFailToReceiveAdWithError:(GADRequestError *)error{
    KLog(@"footer-error:%@",error);
    
}

- (void)adViewWillLeaveApplication:(GADBannerView *)bannerView{
    
}

- (void)layoutSubviews{
    [super layoutSubviews];
    
    self.bannerView.adUnitID = self.adUnitID;
    self.bannerView.frame = self.bounds;
    GADRequest *request = [GADRequest request];
    //    request.testDevices = @[@"2077ef9a63d2b398840261c8221a0c9b"];
    [self.bannerView loadRequest:request];
}


@end

File: iTVLive/iTVLive/Classes/Home/AdmobCell.swift
//
//  AdmobCell.swift
//  iTVshows
//
//  Created by Tengfei on 2017/2/18.
//  Copyright © 2017年 tengfei. All rights reserved.
//

import UIKit
import GoogleMobileAds

class AdmobCell: UITableViewCell,GADBannerViewDelegate {
    @IBOutlet weak var bannerView: GADBannerView!
    
    var rootVc:UIViewController?
    
    class func cellWithTableView(tableView:UITableView) -> AdmobCell{
        let id = "AdmobCell"
        var cell = tableView.dequeueReusableCell(withIdentifier: id) as? AdmobCell
        if cell == nil {
            cell = Bundle.main.loadNibNamed(id, owner: nil, options: nil)?.first as? AdmobCell
        }
        return cell!
    }

    override func awakeFromNib() {
        super.awakeFromNib()
        
//        let deviceId = UIDevice.current.identifierForVendor?.uuidString
        
        print("Google Mobile Ads SDK version: \(GADRequest.sdkVersion())")
        GADRequest.description()
        bannerView.adUnitID = "ca-app-pub-8145075793156354/6341345022"
        bannerView.rootViewController = rootVc
        bannerView.delegate = self
        let request = GADRequest()
//        if deviceId != nil {
//            request.testDevices = [deviceId!]
//        }
        bannerView.load(request)
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        bannerView.rootViewController = rootVc
    }
    
    func adView(_ bannerView: GADBannerView, didFailToReceiveAdWithError error: GADRequestError) {
//        print("--GADBannerView-file:\(error)");
    }
    
    func adViewDidReceiveAd(_ bannerView: GADBannerView) {
//        print("--GADBannerView--success")
    }
}
